packages/cli/src/commands/extensions/disable.ts

```diff
@@ -5,32 +5,31 @@
  */
 
 import { type CommandModule } from 'yargs';
-import { disableExtension } from '../../config/extension.js';
-import { SettingScope } from '../../config/settings.js';
+import { loadSettings, SettingScope } from '../../config/settings.js';
 import { getErrorMessage } from '../../utils/errors.js';
-import { ExtensionEnablementManager } from '../../config/extensions/extensionEnablement.js';
 import { debugLogger } from '@google/gemini-cli-core';
+import { ExtensionManager } from '../../config/extension-manager.js';
+import { requestConsentNonInteractive } from '../../config/extensions/consent.js';
 
 interface DisableArgs {
   name: string;
   scope?: string;
 }
 
 export function handleDisable(args: DisableArgs) {
-  const extensionEnablementManager = new ExtensionEnablementManager();
+  const workspaceDir = process.cwd();
+  const extensionManager = new ExtensionManager({
+    workspaceDir,
+    enabledExtensionOverrides: [],
```

Comment:
It seems a bit odd to be including these values here, no? specifically the enabledExtensionOverrides and requestConsent-- should we make these optional?

Reply:
I went ahead and made `enabledExtensionOverrides` optional but left in `requestConsent` (and the new `requestSetting`).